src/Cli/ConfigGenerator.cs

```diff
@@ -32,24 +32,35 @@ public class ConfigGenerator
         /// </summary>
         public static bool TryGenerateConfig(InitOptions options)
         {
-            if (!TryGetConfigFileBasedOnCliPrecedence(options.Config, out string runtimeConfigFile))
+            string runtimeConfigFile = RuntimeConfigPath.DefaultName;
+            string? environmentValue = Environment.GetEnvironmentVariable(RuntimeConfigPath.RUNTIME_ENVIRONMENT_VAR_NAME);
+            if (!string.IsNullOrWhiteSpace(options.Config))
+            {
+                _logger.LogInformation("Generating user provided config file: {configFileName}", options.Config);
+                runtimeConfigFile = options.Config;
+            }
+            else if (!string.IsNullOrWhiteSpace(environmentValue))
+            {
+                _logger.LogInformation("The environment variable {variableName} has a value of {variableValue}", RuntimeConfigPath.RUNTIME_ENVIRONMENT_VAR_NAME, environmentValue);
+                runtimeConfigFile = RuntimeConfigPath.GetEnvironmentFileName(RuntimeConfigPath.CONFIGFILE_NAME, environmentValue);
+                _logger.LogInformation("Generating environment config file: {config}", runtimeConfigFile);
+            }
+            else
             {
-                runtimeConfigFile = RuntimeConfigPath.DefaultName;
-                _logger.LogInformation($"Creating a new config file: {runtimeConfigFile}");
+                _logger.LogInformation("Generating default config file: {config}", runtimeConfigFile);
             }
 
             // File existence checked to avoid overwriting the existing configuration.
             if (File.Exists(runtimeConfigFile))
             {
-                _logger.LogError($"Config file: {runtimeConfigFile} already exists. " +
-                    "Please provide a different name or remove the existing config file.");
+                _logger.LogError("Config file: {configFileName} already exists. Please provide a different name or remove the existing config file.", runtimeConfigFile);
```

Comment:
This would also benefit from logging the full path of the file name to assist with debugging. If dab cli installed via Nuget, it may not be immediately obvious where the config file is saved if user doesn't provide a full path.